fsspec/core.py

```diff
@@ -498,7 +498,7 @@ def _fetch(self, start, end):
                 self.cache = self.fetcher(start, bend)
                 self.start = start
             else:
-                new = self.fetcher(self.end, bend)
+                new = self.fetcher(self.end, max(end, bend))
```

Comment:
Not sure how this helps, `bend` is either equal to `end`, equal to the file size, or always larger than `end`